CHANGELOG.md

```diff
@@ -1,3 +1,18 @@
+# Master
+
+## Breaking
+
+- `findRecursive` now returns an array instead of an `ArrayElement`.
```

Comment:
This change is necessary because otherwise adding the children to a new array element will change the parent and have undesired effects. I don't see much motivation of find recursive (and parents below) methods for returning `ArrayElement` instead of an array. There is no meta or attributes set, and other high-order functions such as `map` and `reduce` deal in arrays and not ArrayElement.